src/pybind/mgr/dashboard/services/nvmeof_client.py

```diff
@@ -0,0 +1,128 @@
+        response = self.stub.create_bdev(pb2.create_bdev_req(
+            bdev_name=name,
+            rbd_pool_name=rbd_pool,
+            rbd_image_name=rbd_image,
+            block_size=block_size,
+            uuid=uuid
+        ))
+        return json.loads(MessageToJson(response))
+    
+    def resize_bdev(self, name: str, size: int):
+        response = self.stub.resize_bdev(pb2.resize_bdev_req(
+            bdev_name=name,
+            new_size=size
+        ))
+        return json.loads(MessageToJson(response))
+    
+    def delete_bdev(self, name: str, force: bool):
+        response = self.stub.delete_bdev(pb2.delete_bdev_req(
+            bdev_name=name,
+            force=str_to_bool(force)
+        ))
+        return json.loads(MessageToJson(response))
+    
+    def create_subsystem(self, subsystem_nqn: str, serial_number: str, max_namespaces: int,
+                         ana_reporting: bool, enable_ha: bool) :
+        response = self.stub.create_subsystem(pb2.create_subsystem_req(
+            subsystem_nqn=subsystem_nqn,
+            serial_number=serial_number,
+            max_namespaces=int(max_namespaces),
+            ana_reporting=str_to_bool(ana_reporting),
+            enable_ha=str_to_bool(enable_ha)
+        ))
+        return json.loads(MessageToJson(response))
+    
+    def delete_subsystem(self, subsystem_nqn: str):
+        response = self.stub.delete_subsystem(pb2.delete_subsystem_req(
+            subsystem_nqn=subsystem_nqn
+        ))
+        return json.loads(MessageToJson(response))
+    
+    def create_namespace(self, subsystem_nqn: str, bdev_name: str, nsid: int, anagrpid: Optional[str] = None):
+        response = self.stub.add_namespace(pb2.add_namespace_req(
+            subsystem_nqn=subsystem_nqn,
+            bdev_name=bdev_name,
+            nsid=int(nsid),
+            anagrpid=anagrpid
+        ))
+        return json.loads(MessageToJson(response))
+    
+    def delete_namespace(self, subsystem_nqn: str, nsid: int):
+        response = self.stub.remove_namespace(pb2.remove_namespace_req(
+            subsystem_nqn=subsystem_nqn,
+            nsid=nsid
+        ))
+        return json.loads(MessageToJson(response))
+    
+    def add_host(self, subsystem_nqn: str, host_nqn: str):
+        response = self.stub.add_host(pb2.add_host_req(
+            subsystem_nqn=subsystem_nqn,
+            host_nqn=host_nqn
+        ))
+        return json.loads(MessageToJson(response))
+    
+    def remove_host(self, subsystem_nqn: str, host_nqn: str):
+        response = self.stub.remove_host(pb2.remove_host_req(
+            subsystem_nqn=subsystem_nqn,
+            host_nqn=host_nqn
+        ))
+        return json.loads(MessageToJson(response))
+
+    def create_listener(self, nqn: str, gateway: str, trtype: str, adrfam: str,
+                        traddr: str, trsvcid: str):
+        req = pb2.create_listener_req(
```

Comment:
The create_listener is failing currently with following Traceback
```
INFO:control.grpc:Received request to create client.nvmeof.rbd.ceph-node-02.rgfoex   listener for nqn.2016-06.io.spdk:cnode1 at 192.168.105.107:4420., context: <grpc._server._Context object at 0x7ff54dfcd0a0>
[2023-12-14 11:11:03.252679] nvmf_rpc.c: 811:rpc_listen_address_to_trid: *ERROR*: Invalid adrfam: 
ERROR:control.grpc:create_listener failed with: 
 request:
{
  "nqn": "nqn.2016-06.io.spdk:cnode1",
  "listen_address": {
    "trtype": "",
    "traddr": "192.168.105.107",
    "trsvcid": "4420",
    "adrfam": ""
  },
  "method": "nvmf_subsystem_add_listener",
  "req_id": 26
}
Got JSON-RPC error response
response:
{
  "code": -32602,
  "message": "Invalid parameters"
}
```

The request seems ok to me here, `print(req)`
```

nqn: "nqn.2016-06.io.spdk:
gateway_name: "client.nvmeof.rbd.ceph-node-02.
trtype: TCP
adrfam: ipv4
traddr: "192.168.105.107"
trsvcid: "4420"

So not sure why its not mapping values to gateway stub `create_listener`. Can you help with this  issue? @epuertat

Reply:
I attempted to fix it, and I think I was able to do it. I pushed those changes!